common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1743,6 +1744,35 @@ def resource_url(self, resource):
     def publish(self, block, event_type, event):
         pass
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        try:
+            service = super(ModuleSystem, self).service(block=block, service_name=service_name)
+        except NoSuchServiceError:
+            from xmodule.modulestore.django import ModuleI18nService
```

Comment:
Make it simple. @cpennington are you meaning that ? 
```

        service = self._services.get(service_name)
        if callable(service):
            return service(block)
        else:
            return super(ModuleSystem, self).service(block=block, service_name=service_name)

```
sure we need to pass the `ModuleI18nService` with LMS and Studio services.